model/job.go

```diff
@@ -263,7 +326,16 @@ func (j *Job) WriteConfigs(role *Role, lightOpinionsPath, darkOpinionsPath strin
 	if err != nil {
 		return nil, err
 	}
-	config["properties"] = properties
+	config.Properties = properties
+
+	for _, provider := range j.LinkProvides {
+		config.ExportedProperties = append(config.ExportedProperties, provider.Properties...)
+	}
+	for _, consumeTarget := range j.LinkConsumes {
+		if consumeTarget.Role != "" && consumeTarget.Job != "" {
+			config.Consumes[consumeTarget.Name] = consumeTarget
```

Comment:
What happens when `consumeTarget.Name` is `""`?

Reply:
That would mean the provider had no name too, right? (Because otherwise `.Role` and `.Job` wouldn't have been filled in)

I guess we just have a very silly entry… I can skip that case, sure.

Reply:
Yes, the provider must have a name. But do you copy the name into the consumer when you resolve nameless links? (Just asking because it will be faster for you to check, than for me)

Reply:
Yes, we do (and there's tests for it).